userdocs/diagnostics/multiple-class-inheritance.md

```diff
@@ -0,0 +1,45 @@
+# Multiple Inheritance of Classes
+
+In some programming languages, a class can have multiple superclasses. This feature is called multiple inheritance and is unsupported in Swift. However, using protocols, one can achieve a similar effect as multiple inheritance.
+
+For example, the protocol `Animal` and `Flyable` defines blueprints of methods like `move()`. The class `Bird` now provide concrete implementation to its conforming protocols.
+
+```swift
+protocol Flyable {
+    func fly()
+}
+
+protocol Animal {
+    func move()
+}
+
+class Bird: Animal, Flyable {
+    func move() {
+        print("Moving...")
+    }
+    
+    func fly() {
+        print("Flying...")
+    }
+}
+
+```
+A common pattern is when certain functionalities and characteristics are declared in a base class, and a subclass inherit those functionalities and characteristics for free. Swift provide a powerful way for protocols to implement its own requirements through an extension, and these implementations are available to conforming types by default.
+
+Notice how there is no compiler error thrown when class `Bird` does not provide any implementation for `Animal`'s `move()` method? The extension of `Animal` has already provided a default implementation of `move()` that is available for all conforming types.
+  
+```swift
+protocol Animal {
+    func move()
+}
+
+extension Animal {
+    func move() {
+        print("Moving...")
+    }
+}
+```
+
+Conforming types can also provide their own implementations, and these declarations do not need to contain the `override` keyword. When the conforming type is a `class`, note that subclasses of a base class adopting a protocol with default implementations cannot override those declarations without the parent overriding the default implementation. 
```

Comment:
````suggestion
Consider the following example with a class `Bird` conforming to two protocols `Animal` and `Flyable`.

```swift
protocol Flyable {
    func fly()
}

extension Animal {
    func move() {
        print("Moving...")
    }
}

protocol Animal {
    func move()
}

extension Flyable {
    func fly() {
        print("Flying...")
    }
}

class Bird: Animal, Flyable {
    // move() and func fly() use default implementations
}
```

Since `Animal` and `Flyable` provide default implementations for their protocol requirements, `Bird` can conform to both without implementing `fly()` or `move()` and the default implementation will be used. This effect is similar to class inheritance where a subclass inherits properties, subscripts and methods from a superclass. The default implementation can be overriden by defining `fly()` or `move()` without an `override` keyword.

There is one caveat here on the interaction with subclassing. If a superclass conforms a protocol but does not override the default implementation of a protocol requirement, its subclasses cannot override the default implementation ([SR-103](https://bugs.swift.org/browse/SR-103)).
````

Reply:
Feel free to incorporate parts of the suggestion above as you see fit. The main things I would like to emphasize here are:

1. Explanatory text should refrain from trying to "sell" the language. Using phrases like "powerful" and similar isn't very helpful. The reader is already using Swift, and they're trying to figure out how to solve a particular problem.
2. We should cut down on generally anything that is not necessary. Since the main example here is around default implementations, we can reasonably assume that the reader already knows how conformances work normally, so we can get away with one code example instead of two.
3. The part about "When the conforming type is a `class`, note that subclasses of a base class" -- this is actually a gotcha'/road-block, that many people hit (51 people are watching the linked JIRA), so it's better to split it out into its own paragraph instead of continuing in the same paragraph.
4. Linking the JIRA is helpful so that people can easily look things up in case they're curious about why the limitation exists and what has been discussed so far.

Reply:
I applied all your suggestions and agree with the above 3 main points you mentioned. Please feel free to raise more issues and suggestions so I can make sure this note can be improved to its best state before we move forward.